iocore/net/SSLUtils.cc

```diff
@@ -418,17 +306,19 @@ ssl_client_hello_callback(SSL *s, int *al, void *arg)
     return SSL_CLIENT_HELLO_ERROR;
   }
 
-  SSLNetVConnection *netvc = SSLNetVCAccess(s);
-  if (!netvc || netvc->ssl != s) {
-    Debug("ssl.error", "ssl_client_hello_callback call back on stale netvc");
-    return SSL_CLIENT_HELLO_ERROR;
-  }
-
-  bool reenabled = netvc->callHooks(TS_EVENT_SSL_CLIENT_HELLO);
+  SSLNetVConnection *netvc = dynamic_cast<SSLNetVConnection *>(snis);
```

Comment:
Looking over the PR - I am concerned that we are adding 3 dynamic casts.  Dynamic casts seem to be expensive and have shown up in perf top.

Reply:
We talked about the use of dynamic_cast several months ago. I think Alan suggested having `T* NetVConnection::get_service<T>())`. The function would return a pointer for a specified service (SomethingSupport) if the netvc supports it. The downside is that we'd have to maintain the function in addition to having the name in class declaration. And I already stated I'm ok with having the function if dynamic_cast costs much. It just don't have enough priority on anybody's list.